cats-effect/src/main/scala/neotypes/cats/CatsAsync.scala

```diff
@@ -1,10 +1,9 @@
 package neotypes.cats
 
-import cats.{ApplicativeError, effect}
+import cats.effect
 import neotypes.Async
 
-class CatsAsync[F[_]](implicit F: effect.Async[F]) extends Async[F] {
-
+final class CatsAsync[F[_]](implicit F: effect.Async[F]) extends Async[F] {
```

Comment:
Any reason for leaving these classes public?, shouldn't they be private to at least the `neotypes` _package_?

Also, any reason for having named classes at all?, why not simply an **anonymous class** as the body of the `implicit def` ?

Reply:
@BalmungSan good point! I haven't thought about it yet. 
I agree that access should be limited here. Would you like to submit another PR with access fixes? 😀

Reply:
@dimafeng of course, no problem. Will do when I had the time.
However, do you want just to limit the access or going the **anonymous class** route. Which, IMHO, is better.

Reply:
The anonymous class option sounds good to me.